rustc_codegen_spirv/src/codegen_cx/entry.rs

```diff
@@ -241,5 +249,78 @@ impl<'tcx> CodegenCx<'tcx> {
         emit.end_function().unwrap();
 
         emit.entry_point(execution_model, fn_id, name, &[]);
+        
+        entry.execution_modes.iter()
+            .for_each(|(execution_mode, execution_mode_extra)| {
+                emit.execution_mode(fn_id, *execution_mode, execution_mode_extra);
+            });
+    }
+    
+    fn compute_shader_entry_stub(
```

Comment:
There's a decent amount of duplication between this and `shader_entry_stub` (and the existing duplication with kernel too), it might be nice to refactor things into something a little better structured - but probably in a later PR (I can do so), this is fine as-is as a temp solution.